src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -62,11 +94,16 @@ typedef struct {
 
 typedef struct {
     MPIDI_XPMEM_am_unexp_rreq_t unexp_rreq;
+    MPIDI_XPMEM_cnt_t *counter_ptr;
 } MPIDI_XPMEM_am_request_t;
 
 extern MPIDI_XPMEM_global_t MPIDI_XPMEM_global;
 extern MPIR_Object_alloc_t MPIDI_XPMEM_seg_mem;
 
+extern MPIR_Object_alloc_t MPIDI_XPMEM_cnt_mem;
+extern MPIDI_XPMEM_cnt_t MPIDI_XPMEM_cnt_mem_direct[MPIDI_XPMEM_CNT_PREALLOC];
+extern MPIDI_XPMEM_seg_t MPIDI_XPMEM_seg_mem_direct[MPIDI_XPMEM_SEG_PREALLOC];
+
```

Comment:
Is `extern MPIDI_XPMEM_seg_t MPIDI_XPMEM_seg_mem_direct[MPIDI_XPMEM_SEG_PREALLOC];` unneeded?